components/playlist/resources/api/api.ts

```diff
@@ -32,16 +35,22 @@ class PlaylistAPIInstance implements API {
         this.pageHandler.$.bindNewPipeAndPassReceiver())
   }
 
-  getAllPlaylists = async function ():
-      Promise<{playlists: PlaylistMojo.Playlist[]}> {
-        return this.pageHandler.getAllPlaylists()
-      }
+  async getAllPlaylists () {
+    return this.pageHandler.getAllPlaylists()
+  }
 
-  getPlaylist = async function (id: string):
-  Promise<{playlist?: PlaylistMojo.Playlist}> {
+  async getPlaylist (id: string) {
     return this.pageHandler.getPlaylist(id)
   }
 
+  createPlaylist (playlist: PlaylistMojo.Playlist) {
+    this.pageHandler.createPlaylist(playlist)
```

Comment:
It's weird, I think sometimes mojo generates type definitions (not always great ones) but here it doesn't seem to be.

Reply:
Actually, I just had a look at the generated mojom - it looks like mojo did spit the types out into the JS doc but that Typescript seems to be struggling to parse the types when they're multi line.

![image](https://user-images.githubusercontent.com/7678024/182045970-d1dd354c-569d-4c14-8ecf-b219a1f8633d.png)

Kinda unfortunate, but I guess there's not really anything we can do about it :/